src/App.tsx

```diff
@@ -1,39 +1,59 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+interface AppClock {
+  clockName:string,
+  hasClock: boolean,
+}
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, AppClock> {
+  timerId = 0;
+
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
```

Comment:
Create handlers (functions) for the code inside event listeners (you use the same code in both `addEventListener` and `removeEventListener`)